src/Features/Core/Portable/Diagnostics/IDiagnosticAnalyzerService.cs

```diff
@@ -30,17 +30,17 @@ internal interface IDiagnosticAnalyzerService
         /// <summary>
         /// Get specific diagnostics currently stored in the source. returned diagnostic might be out-of-date if solution has changed but analyzer hasn't run for the new solution.
         /// </summary>
-        Task<ImmutableArray<DiagnosticData>> GetSpecificCachedDiagnosticsAsync(Workspace workspace, object id, bool includeSuppressedDiagnostics = false, CancellationToken cancellationToken = default);
+        Task<ImmutableArray<DiagnosticData>> GetSpecificCachedDiagnosticsAsync(Workspace workspace, object id, bool includeSuppressedDiagnostics = false, bool includeNonLocalDocumentDiagnostics = true, CancellationToken cancellationToken = default);
```

Comment:
as per the other thread, i'm ok with the test code being hard to understand.  but i think it's not great that our actual product code gets more confusing with these optional parameters.  I'm not going to block this PR, but i would *reallllllly* like a followup that makes all non-public product code APIs here have no optional parameters.

Reply:
> as per the other thread, i'm ok with the test code being hard to understand. but i think it's not great that our actual product code gets more confusing with these optional parameters. I'm not going to block this PR, but i would _reallllllly_ like a followup that makes all non-public product code APIs here have no optional parameters.

Sounds good. I’ll take a shot at it in this PR for the newly added parameter and guage the cascading impact. If not too much, I’ll do so in this PR, otherwise in a follow-up PR